README.md

```diff
@@ -1,2 +1,202 @@
-# simple-worker
-Teleport Code Challenge
+
+  
+
+# Code Challenge - Teleport
+
+Purpose of this project is to create a small service that is able to `Start`, `Stop`, get `Status` and stream `Output` of a Linux job. The complete package will consist of 3 main components:  
+1. Library to achieve above functionalities (used by server)  
+1. gRPC to achieve communication between clients and the server 
+1. CLI for clients to send requests to server to use above services
+
+
+## Library
+The library will provide functions to create, stop, query status and stream output of a job. It will use `os/exec` package to achieve most of its functionalities.
+
+This library will provide all the utilities needed to `Create`, `Stop`, `GetStatus`, and `Output` of a job. The library will **not** store created jobs internally in memory. It will simply return the `Job` to the user and it is up to the user to manage them. It simplifies the implementation and allows the user to have a flexible job management scheme. However, losing the reference to the job will not allow accessing to that job anymore. I.e, library does not provide any mechanism to list all the existing jobs. However, implementing such mechanism in library in-memory is very simple. This can be achieved by having a global variable of type `map[int]*Job` (protected by a `sync.Mutex`).
+
+### Data structures
+```go
+type Job struct {
+	Id		int64
+
+	cmd    *exec.Cmd
+	status JobStatus	// Status of this Job
+	*sync.Mutex			// to protect Status from simultaneous reads and writes
+	outStream Streamer	// used for creating streaming channel via Output()
+	doneC     chan struct{} // channel for user to block until job exits
+}
+
+type JobStatus {
+	Status string  // string representation of current status
+	Exited bool    // true if job has exited
+	ExitCode int   // exit code of the exited job
+
+	started bool   // true if Start() was successful
```

Comment:
`started` and `Exited` seem like statuses to me. Do you need these fields in addition to `Status` or is there opportunity to combine them?

Reply:
Will combine them into an "enum". No need to have both at the same time.